### ISSUE TITLE
SSL-Setup not working for /share

### ISSUE THREAD

**Issue by apiening** [Alfresco/alfresco-docker-installer#10]
I have answered the question `Do you want to use HTTPs for Web Proxy?` with `yes` at the `alfresco-docker-installer` prompt and selected 443 as the port.
I have pointed my DNS-entry to the server and replaced the localhost-certs in the nginx config folder with my own certificates.
I can  access the base-url, /alfresco, /solr, /api-exlorer and when I access /share I get the error message
```
400 Bad Request
The plain HTTP request was sent to HTTPS port
```
even though I did type the https:// prefix in front of the hostname.

When I prefix the URL I got redirected to (/share/page/) with https, I get the `Alfresco share` login page. When i put in my credentials and select login I get redirected to `/share/page/dologin` but receive another error page there:
```
Something's wrong with this page...
We may have hit an error or something might have been removed or deleted, so check that the URL is correct.
Alternatively you might not have permission to view the page (it could be on a private site) or there could have been an internal error. Try checking with your IT team.
If you're trying to get to your home page and it's no longer available you should change it by clicking your name on the toolbar.
```
Here is what I have set / tried which I think may be relevant:
- I've updated `-Dshare.host=`, `-Daos.baseUrlOverwrite=` for the `alfresco` container and `-Dalfresco.host=` for the `share` container with my hostname.
- I replaced `localhost` for the `proxyName` option in the sed command in `share/Dockerfile` with my FQDN.

Any sort of hint or help is greatly appreciated.

**Comment by aborroy** [comment]
I think you've missed to add `serverName` tag to NGINX configuration.

Add following line in `config/nginx.conf` after `listen` line:

```
server_name alfresco.local;
```

Use your server name instead `alfresco.local`

Additional checks:
* In `docker-compose.yml`: -Dshare.host=alfresco.local -Daos.baseUrlOverwrite=https://alfresco.local/alfresco/aos
* In `share/Dockerfile`: proxyName="alfresco.local"

That should work.

**Comment by apiening** [comment]
Thank you very much @aborroy,
I updated the settings but it did not run on the first try.
Then I entered the share container with `docker-compose exec share bash` and noticed that my `proxyName` was not updated in the tomcat server config (`/usr/local/tomcat/conf/server.xml`).
The reason was simply that I needed to do a `docker-compose build` for the container to be recreated, a `docker-compose down && docker-compose up -d` was not sufficient.
After this it worked wit the settings you suggested.

Thanks again!
I'm closing this issue since it is solved now.
